Source/options.cpp

```diff
@@ -1419,6 +1417,270 @@ uint32_t KeymapperOptions::KeyForAction(string_view actionName) const
 	return SDLK_UNKNOWN;
 }
 
+PadmapperOptions::PadmapperOptions()
+    : OptionCategoryBase("Padmapping", N_("Padmapping"), N_("Padmapping Settings"))
+{
+	buttonToButtonName.emplace(ControllerButton_AXIS_TRIGGERLEFT, "LT");
+	buttonToButtonName.emplace(ControllerButton_AXIS_TRIGGERRIGHT, "RT");
+	buttonToButtonName.emplace(ControllerButton_BUTTON_A, "A");
+	buttonToButtonName.emplace(ControllerButton_BUTTON_B, "B");
+	buttonToButtonName.emplace(ControllerButton_BUTTON_X, "X");
+	buttonToButtonName.emplace(ControllerButton_BUTTON_Y, "Y");
+	buttonToButtonName.emplace(ControllerButton_BUTTON_LEFTSTICK, "LS");
+	buttonToButtonName.emplace(ControllerButton_BUTTON_RIGHTSTICK, "RS");
+	buttonToButtonName.emplace(ControllerButton_BUTTON_LEFTSHOULDER, "LB");
+	buttonToButtonName.emplace(ControllerButton_BUTTON_RIGHTSHOULDER, "RB");
+	buttonToButtonName.emplace(ControllerButton_BUTTON_START, "Start");
+	buttonToButtonName.emplace(ControllerButton_BUTTON_BACK, "Select");
+	buttonToButtonName.emplace(ControllerButton_BUTTON_DPAD_UP, "Up");
+	buttonToButtonName.emplace(ControllerButton_BUTTON_DPAD_DOWN, "Down");
+	buttonToButtonName.emplace(ControllerButton_BUTTON_DPAD_LEFT, "Left");
+	buttonToButtonName.emplace(ControllerButton_BUTTON_DPAD_RIGHT, "Right");
+
+	buttonNameToButton.reserve(buttonToButtonName.size());
+	for (const auto &kv : buttonToButtonName) {
+		buttonNameToButton.emplace(kv.second, kv.first);
+	}
+}
+
+std::vector<OptionEntryBase *> PadmapperOptions::GetEntries()
+{
+	std::vector<OptionEntryBase *> entries;
+	for (auto &action : actions) {
+		entries.push_back(action.get());
+	}
+	return entries;
+}
+
+PadmapperOptions::Action::Action(string_view key, const char *name, const char *description, ControllerButtonCombo defaultInput, std::function<void()> actionPressed, std::function<void()> actionReleased, std::function<bool()> enable, unsigned index)
+    : OptionEntryBase(key, OptionEntryFlags::Invisible, name, description)
+    , defaultInput(defaultInput)
+    , actionPressed(std::move(actionPressed))
+    , actionReleased(std::move(actionReleased))
+    , enable(std::move(enable))
+    , dynamicIndex(index)
+{
+	if (index != 0) {
+		dynamicKey = fmt::format(fmt::runtime(fmt::string_view(key.data(), key.size())), index);
+		this->key = dynamicKey;
+	}
+}
+
+string_view PadmapperOptions::Action::GetName() const
+{
+	if (dynamicIndex == 0)
+		return _(name);
+	dynamicName = fmt::format(fmt::runtime(_(name)), dynamicIndex);
+	return dynamicName;
+}
+
+void PadmapperOptions::Action::LoadFromIni(string_view category)
+{
+	std::array<char, 64> result;
+	if (!GetIniValue(category.data(), key.data(), result.data(), result.size())) {
+		SetValue(defaultInput);
+		return; // Use the default button combo if no mapping has been set.
+	}
+
+	std::string modName;
+	std::string buttonName;
+	for (string_view name : SplitByChar(result.data(), '+')) {

```

Comment:
```suggestion
	for (string_view name : SplitByChar(result, '+')) {
```

Reply:
This actually doesn't work so I left it as-is. `result` is an `std::array<char, 64>` so the call to `result.data()` is necessary to treat it as a C string.